nifi-provenance-reporting-tasks/src/main/java/io/egm/nifi/reporting/EmailProvenanceReporter.java

```diff
@@ -295,21 +329,30 @@ public void indexEvent(final Map<String, Object> event, final ReportingContext c
             getLogger().error("Error sending error email: " + e.getMessage(), e);
         }
     }
-
     public void sendErrorEmail(Map<String, Object> event, ReportingContext context) throws MessagingException {
 
-        String emailSubject = "Error occurred in processor " + event.get("component_name") + " "
-                + "in process group " + event.get("process_group_name");
+        String emailSubject = "[" + context.getProperty(EMAIL_SUBJECT_PREFIX).getValue() + "] : " + "Error occurred in processor " + event.get("component_name") + " "
+                + "in process group " + event.get("process_group_name") +   "from nifi instance: " ;
 
         final Properties properties = new Properties();
         final Session mailSession = this.createMailSession(properties, context);
         final Message message = new MimeMessage(mailSession);
+        InternetAddress[] inetAddressesArray;
+        String specificRecipientAttributeValue = getSpecificRecipientValue(context, event);
```

Comment:
Looking at what you are doing in the `getSpecificRecipientValue` method, you could simplify the checks below

Reply:
I should just check the value of `specificRecipientAttributeValue` and not the value of the attribute name ?

Reply:
Based on what you are doing in `getSpecificRecipientValue`, do you think you need it?

Reply:
nope